### PR TITLE
feat: Add some tsdoc to codegen-only methods

### PR DESCRIPTION
**Pull request by brudil**
Due to some methods only existing within each entities codegen file - rather than in the base entity class - they do not have TSDoc comments, even though these are some of the more essential APIs.

I've added in some basic TSDoc for these which is outputted for each entity. Obviously happy to change up any copy - lagely just defered to the Joist docs using a `@see`.

Unfortunately noisy as a PR due to the codegen changes this results in.

### CODE REVIEW

**Comment by brudil** [comment]
@stephenh Lemme know you if this is something you'll be interested in accepting and I can tidy up the various comments etc

**Comment by stephenh** [comment]
@brudil yep, seems like a good idea! Lmk when you've got it tidied up ; I like the links out to the joist docs!

**Comment by brudil** [comment]
@stephenh Tided things up here, obviously not precious about the content if you have suggestions

File: packages/tests/esm-misc/src/entities/codegen/AuthorCodegen.ts [hunk 1]
```diff
@@ -163,22 +163,66 @@ export abstract class AuthorCodegen extends BaseEntity<EntityManager, string> im
     return getField(this, "updatedAt");
   }
 
+  /**
+   * Partial update taking any subset of the entities fields.
```
Comment (stephenh) [inline packages/tests/esm-misc/src/entities/codegen/AuthorCodegen.ts:167]: Appreciate the invitation to bikeshed! :-)

In that vein, can we add a `\n` after every 1st sentence of the jsdoc, like:

```
/**
 * This is the first sentence.
 *
 * Now more details here.
 */
```

Just an old habit that is engrained. :-)

File: packages/tests/esm-misc/src/entities/codegen/AuthorCodegen.ts [hunk 2]
```diff
@@ -163,22 +163,66 @@ export abstract class AuthorCodegen extends BaseEntity<EntityManager, string> im
...
+   * Unlike `set`, null is used as a marker to mean "unset this field", and undefined
+   * is left as untouched
```
Comment (stephenh) [inline packages/tests/esm-misc/src/entities/codegen/AuthorCodegen.ts:169]: Period after "untouched". And if this is meant as the end of a paragraph, i.e. "Collections" is on a new line, maybe also do a new line between, like:

```
 * is left as untouched.
 *
 * Collections are exha...
```

Assuming the intent was for "Collections" to be a new paragraph (which is fine either way, just wasn't sure intent)

File: packages/tests/esm-misc/src/entities/codegen/AuthorCodegen.ts [hunk 3]
```diff
@@ -163,22 +163,66 @@ export abstract class AuthorCodegen extends BaseEntity<EntityManager, string> im
...
+   * Collections are exhaustively set to the new values, however,
+   * {@link https://joist-orm.io/docs/features/partial-update-apis#incremental-collection-updates | Incremental collection updates} are supported.
```
Comment (stephenh) [inline packages/tests/esm-misc/src/entities/codegen/AuthorCodegen.ts:171]: Again love the doc links :tada: !

File: packages/tests/esm-misc/src/entities/codegen/AuthorCodegen.ts [hunk 4]
```diff
@@ -163,22 +163,66 @@ export abstract class AuthorCodegen extends BaseEntity<EntityManager, string> im
...
     setOpts(this as any as Author, opts);
   }
 
+  /**
+   * Partial update taking any subset of the entities fields.
+   * Unlike `set`, null is used as a marker to mean "unset this field", and undefined
+   * is left as untouched
+   * Collections are exhaustively set to the new values, however,
+   * {@link https://joist-orm.io/docs/features/partial-update-apis#incremental-collection-updates | Incremental collection updates} are supported.
+   * @example
```
Comment (stephenh) [inline packages/tests/esm-misc/src/entities/codegen/AuthorCodegen.ts:192]: Really like the examples!

...following a theme :sweat_smile: , maybe put extra `*` new lines before/after each `@example` block?

File: packages/tests/esm-misc/src/entities/codegen/AuthorCodegen.ts [hunk 5]
```diff
@@ -163,22 +163,66 @@ export abstract class AuthorCodegen extends BaseEntity<EntityManager, string> im
...
+   * });
+   * ```
+   * @see {@link https://joist-orm.io/docs/features/partial-update-apis | Partial Update APIs} on the Joist docs
+   */
   setPartial(opts: PartialOrNull<AuthorOpts>): void {
     setOpts(this as any as Author, opts as OptsOf<Author>, { partial: true });
   }
 
+  /**
+   * Details the field changes of the entity within the current unit of work.
```
Comment (stephenh) [inline packages/tests/esm-misc/src/entities/codegen/AuthorCodegen.ts:207]: Last one I swear, but a newline between the `"unit of work" and the `@see` would be great

File: packages/tests/esm-misc/src/entities/codegen/AuthorCodegen.ts [hunk 6]
```diff
@@ -163,22 +163,66 @@ export abstract class AuthorCodegen extends BaseEntity<EntityManager, string> im
...
+   * @see {@link https://joist-orm.io/docs/features/changed-fields | Changed Fields} on the Joist docs
+   */
   get changes(): Changes<Author> {
     return newChangesProxy(this) as any;
   }
 
+  /**
+   * Traverse from this entity using a lens, and load the result
```
Comment (stephenh) [inline packages/tests/esm-misc/src/entities/codegen/AuthorCodegen.ts:215]: Period at "result.", ideally for all of the "first sentence overviews".

**Review by stephenh — APPROVED** [review]
This is great! As invited, I added a few asks, all just formatting related, thought the content was all great/well-written. Thank you!

**Comment by brudil** [comment]
Pushed with your feedback!

**Pull request merged**